src/transformers/modeling_flash_attention_utils.py

```diff
@@ -182,7 +182,8 @@ def _flash_attention_forward(
     flash_kwargs = {"window_size": (sliding_window, sliding_window)} if use_sliding_windows else {}
 
     if is_flash_attn_greater_or_equal("2.4.1"):
-        flash_kwargs["deterministic"] = deterministic
+        is_deterministic = os.environ.get("FLASH_ATTENTION_DETERMINISTIC", "0") == "1"
+        flash_kwargs["deterministic"] = is_deterministic
```

Comment:
If this is the case, then the `deterministic` flag has no effect. It should either be accounted for or removed

Reply:
```suggestion
    if is_flash_attn_greater_or_equal("2.4.1"):
        if deterministic is None:
            deterministic = os.environ.get("FLASH_ATTENTION_DETERMINISTIC", "0") == "1"
        flash_kwargs["deterministic"] = deterministic
```

Would it be better to change the default value of the deterministic option to None and modify the code as shown above?

Reply:
Yep - LGTM!

Reply:
Thank you so much, @amyeroberts , for your help with this pull request!